R/package.R

```diff
@@ -50,6 +50,10 @@
 argufy <- function(fun, ...) {
   if (!is.function(fun)) stop("'fun' must be a function")
 
+  # these statements are needed to get S4 functions to work properly
+  was_s4 <- isS4(fun)
+  old_attributes <- attributes(fun)
+
```

Comment:
Agreed I was confused as well, easy way to see the real function definition
is `body(fun)` or `print(fun, useSource = FALSE)`

On Tue, Aug 11, 2015 at 4:41 PM, Gábor Csárdi <notifications@github.com>
wrote:

> In R/package.R
> <https://github.com/gaborcsardi/argufy/pull/5#discussion_r36795931>:
>
> > @@ -50,6 +50,10 @@
> >  argufy <- function(fun, ...) {
> >    if (!is.function(fun)) stop("'fun' must be a function")
> >
> > +  # these statements are needed to get S4 functions to work properly
> > +  was_s4 <- isS4(fun)
> > +  old_attributes <- attributes(fun)
> > +
>
> If the code can be printed easily then it is fine as it is now. I'll just
> add it to the docs. This confused me until I debugged argufy(), because
> it did not seem to do anything. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gaborcsardi/argufy/pull/5/files#r36795931>.
>